### DISCUSSION TITLE
server list type issue, and other questions

### DISCUSSION THREAD

**Discussion by Tr3yWay996** [LOENS2/pterodactyl_exporter#41] [Q&A] [answered]
As the tittle say, I do not understand how It's supposed to work even after having everything setup like said in the "README" I tried to input http://localhost:6060 it wasn't working, so I gave it the whole public ipv4 then and another issue come up as seen in the screenshot
![image](https://github.com/user-attachments/assets/be6459c0-1b33-4f44-a910-361277157e30)

**Comment by LOENS2** [comment]
Could you please elaborate further on how your setup exactly looks like.

The basic steps are:

1. Install the exporter (you already did that)
2. Add the exporter to the Prometheus yml config file (the ip has to be the physical machine's or the vm's ip)
3. Add the Prometheus server to Grafana (I assume, that you've done that, too)
4. Import the dashboard to Grafana.

Btw: The Prometheus standard port is 9090 unless you've changed it.
The port in the prometheus config obviously has to be the same as the one of your pterodactyl exporter container.

**Reply by Tr3yWay996** [reply]
So I installed Grafana yesterday as system service, today I wanted to install Ptero Exporter on pterodactyl via the egg, have set up port 6060 no HTTPS no SSL since it doesn't work with it and the part that I don't understand well is with Prometheus

**Reply by Tr3yWay996** [reply]
Will try something and uhh i will let you know

**Reply by LOENS2** [reply]
Ok, now I get what your problem is. You can't set the Grafana datasource straight to the exporter. You have to set up Prometheus and as you've correctly pointed out, they only provide the pre-compiled binary on their site. That is why I recommend running Prometheus as a docker container as well as grafana and the pterodactyl exporter, because keeping them up to date is much easier. Now, within Grafana, you have to add the Prometheus instance as a datasource and then you are able to pull the metrics.

**Reply by Tr3yWay996** [reply]
So I should delete Grafana and put it into a docker ? Hmm, alright I will do, same for Prometheus then and i will come back
Btw I deleted the edit history (won't redo, it's useless) because I thought I was completely wrong about the binary, but alr apparently I wasn't

**Reply by Tr3yWay996** [reply]
New thing is that port 9090 used by Ptero export cannot be used by Prometheus as you can see here, but if I should be using the same port then what do i do ? because its not possible, i missed something i know but i cant point what exactly
![image](https://github.com/user-attachments/assets/3e23a0ab-b767-48d9-8dbe-1f7b6686bc8f)

**Reply by Tr3yWay996** [reply]
I change ptero exporter port to 9531 and kept 9090 for Prometheus, will try it out now

**Comment by Tr3yWay996** [comment]
Okay so uhh I have "NO DATA" at every panel, what else is needed as it should have been done already, but apparently it's not.
I properly made a user account for the exporter, gave it read perm to the backup onto the server I want to monitor, I have set up Grafana Prometheus and everything and I still have no data anywhere

**Comment by LOENS2** [comment]
Well. There are a few steps you can try in order to debug the problem.

1. Try to curl the pterodactyl exporter. It should return all the servers that are covered by your config and show their metrics

2. You can open the prometheus web ui in the browser and query the metri s this way.

3. Try to query the exporter metrics within Grafana. This can be done with the explore feature.

If one of the above steps fails, try to mitigate the cause and you should get data in zhe charts and graphs.

**Reply by Tr3yWay996** [reply]
> Try to curl the pterodactyl exporter. It should return all the servers that are covered by your config and show their metrics

This gives me this big chunk of text and that's it, nowhere do I see the servers it's active on, so I don't understand what you are asking me, apart if It's that ??
![image](https://github.com/user-attachments/assets/05178d05-ea67-4ba5-80e9-2c3f846e164b)

**Reply by LOENS2** [reply]
That is exactly what I meant with the first point. For some reason the exporter doesn't get any data from pterodactyl. That means, that you either misconfigured the exporter, your account doesn't have access to any server or the exporter isn't able to connect to the panels api. The latter ones should produce a warning or error message in the exporters logs.

**Reply by Tr3yWay996** [reply]
1st  The exporter is configured like it should be, on the right port, no HTTPS, ignore SSL set to 0 and so on
![image](https://github.com/user-attachments/assets/cf55b5a5-fba5-4627-97ba-095a077dd470)
2nd the account used definitely have access to the server I want to monitor
![image](https://github.com/user-attachments/assets/c165c40c-fe9a-41a1-97a6-b8945d67eefa)
![image](https://github.com/user-attachments/assets/efa12668-9caf-41d4-bf98-e18e95cca9d5)

3rd  It can access the API without issues, it even reports it and there is no sign of it not reaching the API
![image](https://github.com/user-attachments/assets/58f4e33b-280c-4bac-a7ee-fbfb95e4dc0d)

**Reply by Tr3yWay996** [reply]
Also the API key is clearly being used
![image](https://github.com/user-attachments/assets/84d9fe02-4295-41a9-8650-5adeb22825a4)

**Reply by LOENS2** [reply, marked as answer]
That all looks fine to me except the "admin-all" server list type. Is your pterodactyl account actually an admin account? If you have set the permissions on the individual servers then it is sufficient to set the server list type to an empty string.

**Reply by Tr3yWay996** [reply]
its not an admin acount no i will try the empty string tho

**Reply by Tr3yWay996** [reply]
Nevermind it just was very slow to update, dang it was that all along ? argh
Well thanks a lot for the support.
edit: Now i have a question tho about the graph's do they reset each time i restart the exporter ? Because the uptime is here yes but everything else is empty pretty much

**Reply by LOENS2** [reply]
You're welcome.

**Reply by Tr3yWay996** [reply]
Oh btw, another question, does the data show up the same as how i see it in the panel ? But just extended log time and better graphs
because i got 3 core and i know pterodactyl and wings count in 100% per said cores, its the same here ? I guees yes since it kinda looks like what i see on the panel on itself

**Reply by LOENS2** [reply]
The graphs don't reset when you restart the exporter. The data is saved by the prometheus server. Just be sure to have a persistent volume mounted to the container so the data doesnt get deleted when you restart it.

**Reply by LOENS2** [reply]
The data is the exact same as shown in the panel. I don't particuarly like the representation of the multi core usage either, but that is a limitation of pterodactyl and there is no way of getting a better representation with data provided by the api.

**Reply by Tr3yWay996** [reply]
Hmm i see, well for the perssistent storage it should be since i followed their doc about how to create one and run it with it

**Reply by Tr3yWay996** [reply]
> The data is the exact same as shown in the panel. I don't particuarly like the representation of the multi core usage either, but that is a limitation of pterodactyl and there is no way of getting a better representation with data provided by the api.

It's fine, I found it weird at first the 100% per core system, but we get used to it quickly but i mostly wanted to know since it was showing 46% and I thought 'oh no not another issue, 46% of the 3 cores isn't used' so i wanted to be sure thanks

**Comment by Tr3yWay996** [comment]
Hey, is it normal to have the network but not this traffic graph running ? When I say normal, I mean, does it need any other setup ?
![image](https://github.com/user-attachments/assets/2f29e49d-ee0e-4ca9-8358-033c40c83aa6)
